### PR TITLE
Add topic producer API Python side

### PR DESCRIPTION
**Pull request by cbornet**
See #649 

After that, remaining work for topic producer will be:
* Support AVRO schema
* Add e2e tests

### CODE REVIEW

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_grpc/api.py [hunk 1]
```diff
@@ -63,12 +63,28 @@ def headers(self) -> List[Tuple[str, Any]]:
...
 
 
+class TopicProducer(ABC):
+    """The topic producer interface"""
+
+    async def write(self, topic: str, record: Record):
+        """Write a record to a topic (for async methods)."""
+        pass
+
+    def write_future(self, topic: str, record: Record) -> Future:
```
Comment (cbornet) [inline langstream-runtime/langstream-runtime-impl/src/main/python/langstream_grpc/api.py:73]: Suggestions for a better name ?

Comment (eolivelli) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_grpc/api.py:73]: what about "awrite" for the async write and "write" for the sync version ?

Comment (cbornet) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_grpc/api.py:73]: Yes, I've thought of that. Now both methods are kinda "async", one with async/await, the other with Future.

Comment (cbornet) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_grpc/api.py:73]: But let's use `awrite` as it's a defacto convention when 2 methods do the same thing and one does it on the event loop.

**Pull request merged**